proteuslib/unit_models/reverse_osmosis_1D.py

```diff
@@ -0,0 +1,800 @@
+            except TypeError:
+                raise ConfigurationError("RO model only supports one solvent and one or more solutes,"
+                                         "the provided property package has specified a component '{}' "
+                                         "that is not a solvent or solute".format(c))
+        if len(self.solvent_list) > 1:
+            raise ConfigurationError("RO model only supports one solvent component,"
+                                     "the provided property package has specified {} solvent components"
+                                     .format(len(self.solvent_list)))
+
+        if self.config.transformation_method is useDefault:
+            _log.warning(
+                "Discretization method was "
+                "not specified for the "
+                "reverse osmosis module. "
+                "Defaulting to finite "
+                "difference method."
+            )
+            self.config.transformation_method = "dae.finite_difference"
+
+        if self.config.transformation_scheme is useDefault:
+            _log.warning(
+                "Discretization scheme was "
+                "not specified for the "
+                "reverse osmosis module."
+                "Defaulting to backward finite "
+                "difference."
+            )
+            self.config.transformation_scheme = "BACKWARD"
+
+
+    def build(self):
+        """
+        Build 1D RO model (pre-DAE transformation).
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        # Call UnitModel.build to setup dynamics
+        super().build()
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        self.solvent_list = Set()
+        self.solute_list = Set()
+        self._process_config()
+
+        # Build 1D Control volume for feed side
+        self.feed_side = ControlVolume1DBlock(default={
+            "dynamic": self.config.dynamic,
+            "has_holdup": self.config.has_holdup,
+            "area_definition": self.config.area_definition,
+            "property_package": self.config.property_package,
+            "property_package_args": self.config.property_package_args,
+            "transformation_method": self.config.transformation_method,
+            "transformation_scheme": self.config.transformation_scheme,
+            "finite_elements": self.config.finite_elements,
+            "collocation_points": self.config.collocation_points
+        })
+
+        feed_side = self.feed_side
+        # Add geometry to feed side
+        feed_side.add_geometry()
+        # Add state blocks to feed side
+        feed_side.add_state_blocks(has_phase_equilibrium=False)
+        # Populate feed side
+        feed_side.add_material_balances(balance_type=self.config.material_balance_type,
+                                        has_mass_transfer=True)
+        feed_side.add_energy_balances(balance_type=EnergyBalanceType.none)
```

Comment:
As noted above.

However, also note that the underlying bug in the ideal property package was fixed such that you should be able to do an energy balance now and not see excessive temperature change due to a change in pressure.